processing/src/main/java/io/druid/query/Query.java

```diff
@@ -61,8 +63,9 @@
   String SELECT = "select";
   String TOPN = "topN";
   String DATASOURCE_METADATA = "dataSourceMetadata";
+  String JOIN = "join";
 
-  DataSource getDataSource();
+  Iterable<DataSourceWithSegmentSpec> getDataSources();
```

Comment:
Since every query type must have only one data source before, this API returns a dataSource. However, now a query can have one or multiple data sources. I think this API should be changed because ```Query``` should abstract every type of queries.